### DISCUSSION TITLE
Is there an unique ID of ESP which can be used for serial number purposes

### DISCUSSION THREAD

**Discussion by benjaminnestler** [arendst/Tasmota#20984] [Support request] [answered]
Hi everyone,

I have a quick question. Is it possible to read a unique ID from the ESP chipset using an existing Tasmota function that can be used as a serial number for devices? Ideally, it would be great if this unique ID could also be queried through one of the Tasmota interfaces already available.

**Comment by sfromis** [comment, marked as answer]
The "MAC Address" visible on the "Information" page should be an unique id. This is also available from the command "Status 4". In Berry it is included in the result of `tasmota.wifi()`, and the different MAC for Ethernet from `tasmota.eth()`. The "Information" page also has a field "ESP Chip Id", but that is just the last 3 bytes of the MAC, and therefore not as unique.

**Reply by benjaminnestler** [reply]
Hi @sfromis,

As you mentioned, the MAC address should be a unique ID, but unfortunately, that's not always the case in reality. Would you consider this an acceptable risk?

**Reply by sfromis** [reply]
It tells nothing to refer to the vague term "reality", instead of looking at the specific context. Do you have cases of two Espressif chips having the same MAC?

Sure, there can be other cases where users are faking a MAC, but you asked about "ESP chipset" instead of in general.

There is a real risk that the "ESP Chip Id" is not unique, as this is only half of the MAC.

Anyway, there is no other unique ESP chip "serial number" than the MAC for the network interface you use.
